ui/lib/kv/addon/components/page/list.hbs

```diff
@@ -5,79 +5,107 @@
   </:tabLinks>
 
   <:toolbarFilters>
-    <KvListFilter
-      @model={{@model}}
-      @mountPoint={{this.mountPoint}}
-      @filterValue={{@filterValue}}
-      @pageFilter={{@pageFilter}}
-    />
+    {{#unless @noMetadataListPermissions}}
+      <KvListFilter
+        @model={{@model}}
+        @mountPoint={{this.mountPoint}}
+        @filterValue={{@filterValue}}
+        @pageFilter={{@pageFilter}}
+      />
+    {{/unless}}
   </:toolbarFilters>
 
   <:toolbarActions>
     {{! TODO this create link route needs to handle creating from a directory e.g. beep/boop/  "initialKey"}}
     <ToolbarLink @route="create" @type="add">Create secret</ToolbarLink>
   </:toolbarActions>
 </KvPageHeader>
-{{! TODO add permissions on the menu items }}
-{{#if @model.secrets}}
-  {{#each @model.secrets as |secret|}}
-    <LinkedBlock
-      class="list-item-row"
-      @params={{array (if secret.pathIsDirectory "list-directory" "secret.details") secret.fullSecretPath}}
-      @linkPrefix={{this.mountPoint}}
-    >
-      <div class="level is-mobile">
-        <div class="level-left">
-          <div>
-            <Icon @name="user" class="has-text-grey-light" />
-            <span class="has-text-weight-semibold is-underline">
-              {{secret.path}}
-            </span>
-          </div>
+
+{{#if @noMetadataListPermissions}}
+  <div class="box is-fullwidth is-shadowless has-tall-padding">
+    <div class="selectable-card-container one-card">
+      <OverviewCard @cardTitle="View secret" @subText="Type the path of the secret you want to read.">
+        <div class="has-top-margin-m is-flex">
+          <InputSearch
```

Comment:
Awesome! I didn't know we had an `<InputSearch>` component :D

Reply:
I totally forgot about it until I saw it in the get-credentials-component!